packages/wallet-sdk/src/CoinbaseWalletProvider.ts

```diff
@@ -54,6 +54,13 @@ export class CoinbaseWalletProvider extends ProviderEventEmitter implements Prov
             storeSignerType(signerType);
             break;
           }
+          case 'wallet_connect': {
+            const signer = this.initSigner('scw');
+            this.signer = signer;
+            await signer.handshake({ method: 'handshake' }); // exchange session keys
+            const result = await signer.request(args); // send diffie-hellman encrypted request
+            return result as T;
+          }
```

Comment:
we might need to move the `this.signer = signer` to be after `signer.request`, if user reject this request, the signer won't be assigned

Reply:
updated